toolchain/check/handle_impl.cpp

```diff
@@ -202,12 +202,18 @@ static auto PopImplIntroducerAndParamsAsNameComponent(
       context.node_stack().PopWithNodeIdIf<Parse::NodeKind::ImplForall>();
 
   if (implicit_param_patterns_id) {
-    // Emit the `forall` match. This shouldn't produce any `Call` params,
+    // Emit the `forall` match. This shouldn't produce any valid `Call` params,
     // because `impl`s are never actually called at runtime.
     auto call_params_id =
         CalleePatternMatch(context, *implicit_param_patterns_id,
                            SemIR::InstBlockId::Invalid, SemIR::InstId::Invalid);
-    CARBON_CHECK(call_params_id == SemIR::InstBlockId::Empty);
+    CARBON_CHECK(call_params_id == SemIR::InstBlockId::Empty ||
+                 context.inst_blocks()
+                     .Get(call_params_id)
+                     .drop_while([](SemIR::InstId inst_id) {
+                       return inst_id == SemIR::ErrorInst::SingletonInstId;
+                     })
+                     .empty());
```

Comment:
```suggestion
                 llvm::count_if(context.inst_blocks().Get(call_params_id),
                                [](SemIR::InstId inst_id) {
                                  return inst_id !=
                                         SemIR::ErrorInst::SingletonInstId;
                                }) == 0);
```

Was wondering if a count-based version might be appealing as a way to avoid the intermediary arrayref? Though this only happens on invalid code, so it seems minor -- I think this is basically whichever approach you consider more readable.

Reply:
Ooh, better:

```suggestion
                 llvm::all_of(context.inst_blocks().Get(call_params_id),
                              [](SemIR::InstId inst_id) {
                                return inst_id ==
                                       SemIR::ErrorInst::SingletonInstId;
                              }));
```